src/server/internalServer.cpp

```diff
@@ -759,14 +759,25 @@ std::unique_ptr<Response> InternalServer::handle_viewer_settings(const RequestCo
   return ContentResponse::build(*this, RESOURCE::templates::viewer_settings_js, data, "application/javascript; charset=utf-8");
 }
 
+std::string userLang = "eng";
```

Comment:
A mutable global variable is an issue for concurrent access.

Reply:
Made it a member of internalServer, i've a feeling there are better ways but if this works then nice!

Reply:
The more general problem for concurrent access is a shared mutable variable. A global variable merely has more chances of becoming one. Your "fix" doesn't prevent m_userLang from being changed from two or more different threads during concurrent execution of multiple `/nosj` requests. However, it is not clear why you needed to make `userLang` a data member of `InternalServer` at all. A local variable should be just fine.

Since I haven't yet looked at your other changes, please correct this in place (i.e. without introducing a fixup commit).

Reply:
Without having some sort of "shared" access to m_userLang between HTMLDumper & InternalServer - the `/nojs/download` page wouldn't know what userlang to display. Hence the requirement of a member `m_userLang`.

Consider this:

`/nojs?userlang=test`
Click a book's download button & you're redirected to `/nojs/download/xyz` - if we're using a local variable, it would ask for `userlang` argument again - not get it & set the language to default English.


Another solution is probably setting all download links to have userlang as query. Wouldn't require any member in internalServer then, is this better?

Reply:
You must not store any state *in the server* between requests (in this case, with proper timing `m_userLang` set by one user's access to `/nojs` may affect the language used for serving `/nojs/download/xyz` to another user). 

Note that the `userlang` URL query parameter is only one way of specifying the user language. The user-language (no matter where it comes from) is stored in a cookie:
https://github.com/kiwix/libkiwix/blob/88de978a9c52aa8e64dad883e949be1e9250dfa7/src/server/request_context.cpp#L210-L228
https://github.com/kiwix/libkiwix/blob/88de978a9c52aa8e64dad883e949be1e9250dfa7/src/server/response.cpp#L390-L395

Reply:
Done